script.c

```diff
@@ -16,78 +16,90 @@
 
 #include "script.h"
 #include "plostd.h"
+#include "phfs.h"
 #include "cmd.h"
 #include "hal.h"
+#include "errors.h"
 
 
 /* Linker symbol points to the beginning of .data section */
 extern char script[];
 
 
 struct {
-	u32 cmdCnt;
-	char basicCmds[MAX_SCRIPT_LINES_NB][LINESZ];
+	u32 cnt;
+	char cmds[MAX_SCRIPT_LINES_NB][LINESZ];
 } script_common;
 
 
+static int script_regFile(char *cmd)
+{
+	int i;
+	size_t sz = 0;
+	addr_t addr = 0;
+	unsigned int pos = 0;
+	char alias[3][LINESZ + 1];
+
+	for (i = 0; i < 3; ++i) {
+		if (cmd_getnext(cmd, &pos, "(:) \t", NULL, alias[i], sizeof(alias[i])) == NULL || *alias[i] == 0)
+			break;
+
+		if (i == 1) {
+			if (plostd_ishex(alias[i]) < 0){
+				return ERR_ARG;
+			}
+			addr = plostd_ahtoi(alias[i]);
+		}
+		else if (i == 2) {
+			if (plostd_ishex(alias[i]) < 0) {
+				return ERR_ARG;
+			}
+
+			sz = plostd_ahtoi(alias[i]);
+
+		}
```

Comment:
This can be simplified to something like
```suggestion
		if (i == 0)
			continue;

		if (plostd_ishex(alias[i]) < 0){
			return ERR_ARG;

		if (i == 1) {
			addr = plostd_ahtoi(alias[i]);
		}
		else if (i == 2) {
			sz = plostd_ahtoi(alias[i]);
		}
```

Reply:
Done